test/test_jujucontext.py

```diff
@@ -14,69 +14,72 @@
 
 from __future__ import annotations
 
+import os
 from pathlib import Path
 
-from ops.jujucontext import _JujuContext
+import pytest
+
+from ops.jujucontext import JujuContext
 from ops.jujuversion import JujuVersion
 
 
 class TestJujuContext:
     def test_both_str_and_int_fields_default_to_none(self):
-        juju_context = _JujuContext.from_dict({'JUJU_VERSION': '0.0.0'})
+        juju_context = JujuContext._from_dict({'JUJU_VERSION': '0.0.0'})
```

Comment:
It seems that `JujuContext.from_environ(some_dict)` is the new way to using the context, would it be better to test this new way of creating the context over the helper function?

Reply:
Alternatively, maybe some of these tests can be retired if the same functionality is covered by new tests below?

Reply:
We're still using both at the moment (see the other thread with Ben and James). I think while they are both explicitly used, we should have some tests that explicitly test each of them, rather than relying on one being indirectly called.